applications/nrf_desktop/src/hw_interface/leds.c

```diff
@@ -195,13 +237,36 @@ static bool event_handler(const struct event_header *eh)
 
 		struct led *led = &leds[event->led_id];
 
-		led->effect = event->led_effect;
+		__ASSERT_NO_MSG(event->led_effect);
 
-		if (initialized) {
-			led_update(led);
-		}
+		if (led->stream && !event->stream) {
+			LOG_INF("Save no-stream effect");
 
-		return false;
+			led->state_effect = event->led_effect;
+		} else if (led->stream && event->stream) {
+			LOG_INF("Next stream event");
+
+			enqueue_stream_effect(event);
+		} else if (!led->stream && event->stream) {
+			LOG_INF("First stream event, save no-stream effect");
+
+			enqueue_stream_effect(event);
```

Comment:
the effect will be replayed twice as it was hooked both as current and next effect

Reply:
done